Cython/Compiler/ExprNodes.py

```diff
@@ -15221,11 +15221,12 @@ class AssignmentExpressionNode(ExprNode):
     subexprs = ["rhs"]
     child_attrs = ["rhs", "assignment"]  # This order is important for control-flow (i.e. xdecref) to be right
 
-    is_temp = False
-    assignment = None
-    clone_node = None
+    is_temp: bool = False
+    rhs: ExprNode
+    assignment: SingleAssignmentNode
+    is_literal_assignment: bool = False
 
-    def __init__(self, pos, lhs, rhs, **kwds):
+    def __init__(self, pos, lhs: NameNode, rhs: ExprNode, **kwds):
         super().__init__(pos, **kwds)
         self.rhs = ProxyNode(rhs)
         assign_expr_rhs = CloneNode(self.rhs)
```

Comment:
Do we need to wrap the nodes at this early stage, or can we postpone it until `analyse_types()`, to see if we actually need to do it?
Also, would it help to move the ownership into the assignment node at this stage, and take back the RHS reference from there in `analyse_types()`?

I'm just trying to think of ways to avoid the back and forth un-wrapping into auxiliary nodes that are only *sometimes* needed and could possibly *only* be done in the respective cases.

Reply:
I think I've refactored it to defer it.